Background layers processor

jesseeichar opened the pull request:
Added Processor for adding background layers to a map. See #231 for details.

tsauerwein commented:
I added some comments, but this is looking good!

Will this PR also close #230? Or do you still want something generic (for what use-case)?

sbrunner commented:
Looks good but shouldn't he also be able to add overlay ?
Sometimes we want to add a graticule, watermark, or something else generated by the WMS server, ...

jesseeichar commented:
I have updated this PR so that it can do overlays or background layers.  But unfortunately I rebased onto develop and lost the comments of @tsauerwein 

@tsauerwein do you remember your comments?  Or perhaps you can re-review this?  I need this for the sogelink.

sbrunner commented on core/src/main/java/org/mapfish/print/processor/map/AddStaticLayersProcessor.java:
```diff
@@ -0,0 +1,107 @@
+        /**
+         * The attribute containing the static layers to add to the map.
+         */
+        public StaticLayersAttribute.StaticLayersAttributeValue staticLayers;
```
`backgroundLayers` isn't better than `staticLayers`? (for the attribute name)

jesseeichar replied:
It is not necessarily background layers.  It could be overlay layers

sbrunner replied:
OK, thanks :-)

tsauerwein commented on core/src/main/java/org/mapfish/print/processor/map/AddStaticLayersProcessor.java:
```diff
@@ -0,0 +1,107 @@
+/**
+ * This processor adds the configured set of layers to the map.
+ *
+ * This is useful when all maps should have a default set of background layers or overlayers
```
"... overlays"

tsauerwein commented on core/src/main/java/org/mapfish/print/wrapper/PJoinedArray.java:
```diff
@@ -0,0 +1,118 @@
+
+    @Override
+    public PObject getObject(final int i) {
+        int index = i;
```
Could you also use `get()` for this method? Same for `getArray`.

jesseeichar replied:
That would require an API change of an API that goes back to V1.  I'd rather leave the API alone.  Or if we want to change it we can do it in a separate PR.

tsauerwein replied:
Maybe I am missing something, but wouldn't `return (PObject) get(i);` do the same thing?

jesseeichar replied:
Ah,  I misunderstood the question.  I think in some cases it will not return the same thing but I am not sure.  Its worth looking into a bit more.

tsauerwein commented:
My original comment (which was lost) was about the dependencies in `mapfish-spring-processors.xml`, which you addressed already. So it looks good to me.
